### PR TITLE
CERT-6637: Add entry about receiveOrFallback in glossary

### PR DESCRIPTION
**Pull request by christiane-certora**
Based on changes in https://github.com/Certora/EVMVerifier/pull/6226 for the display of that.

Naming convention:
 - PRs for features that are in design should have the "proposal" label
 - PRs for features that haven't landed yet should have the "future" label
 - PRs for upcoming releases should have the "release" label
 - PRs with new documentation for existing features should have the "existing feature" label

Before requesting review:
 - Make sure there is a ticket in the DOC board in Jira
 - Make sure CI is passing
   - Spell check failure may require adding backticks around code or updating `spelling_wordlist.txt`
   - See `README.md` for information about style and markdown syntax
   - If the CI Details link gives a 404, you need to log in to readthedocs.com
 - Add link to generated documentation here
   - you can find this by following the read the docs link from the CI check
 - Ask for help in #documentation

Jira ticket: TODO
Link to generated documentation: https://certora-certora-prover-documentation--284.com.readthedocs.build/en/284/docs/user-guide/glossary.html#term-receiveOrFallback

### CODE REVIEW

File: docs/user-guide/glossary.md [hunk 1]
```diff
@@ -181,6 +181,12 @@ quantified expression
   referred to as *quantified expressions*.  See {ref}`logic-exprs` for
   details about quantifiers in CVL.
 
+receiveOrFallback
+  A special function we introduce in every contract to model the behavior of solidity
+  for calls with no data or that do not resolve to any contract function.
```
Comment (johspaeth) [inline docs/user-guide/glossary.md:186]: I think we should we add a link to the official EVM docs here?

**Review by johspaeth — CHANGES_REQUESTED** [review]
Just a minor point. I think we should add a link. Thanks.

File: docs/user-guide/glossary.md (hunk 1 above)
Comment (christiane-certora) [reply docs/user-guide/glossary.md:186]: Something like this? https://docs.soliditylang.org/en/latest/contracts.html#fallback-function

Comment (johspaeth) [reply docs/user-guide/glossary.md:186]: Exactly 👍

**Review by johspaeth — APPROVED** [review]

**Comment by johspaeth** [comment]
Thanks for preparing - I added the label `release`. Uri or me will merge this with the next release then such that docs align with the current version.

**Pull request merged**